src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/HeapHprofBinWriter.java

```diff
@@ -419,9 +419,7 @@ public synchronized void write(String fileName) throws IOException {
                     }
                 };
             }
-            hprofBufferedOut = new SegmentedOutputStream(dataOut);
-        } else {
-            hprofBufferedOut = new SegmentedOutputStream(fos, false /* allowSegmented */);
+            hprofBufferedOut = dataOut;
```

Comment:
I do not understand why the local variable `dataOut` is needed.
The code will be simpler without it:
```
    OutputStream dataOut = fos;
     hprofBufferedOut = dataOut;
     if (useSegmentedHeapDump) {
         if (isCompression()) {
             dataOut = new GZIPOutputStream(fos) {
                 {
                     this.def.setLevel(gzLevel);
                 }
             };
         }
             hprofBufferedOut = dataOut;
     }
```
=====>
```
     hprofBufferedOut = fos;
     if (useSegmentedHeapDump) {
         if (isCompression()) {
             hprofBufferedOut = new GZIPOutputStream(fos) {
                 {
                     this.def.setLevel(gzLevel);
                 }
             };
         }
     }
```